### DISCUSSION TITLE
Will Remix become "batteries included" in the future?

### DISCUSSION THREAD

**Discussion by danielbergholz** [remix-run/remix#7731] [General]
Hello friends! 

First of all, I wanted to say I'm extremely happy with Remix, and I'm seeing multiple frontend (and even backend) developers choosing it as the default fullstack framework. It fills the gap between having a complex and rich UI with React, but also have access to the server, without the need to build a REST API in the middle.

There is still one thing that I personally miss on Remix, and that is the "batteries included" philosophy from Ruby on Rails. Rails comes with ORM with validations, background jobs, sending and receiving emails, file uploads, testing, localization, and even REPL. Remix gives you views (components) and controllers (loader and action functions), the rest is up to you to set up. I know that we have the stacks, but they are templates, you still need to manage all the "glue code" from the libraries used and make sure they work together well.

Are there plans for Remix to become "batteries included" in the future? Maybe by having native packages for authentication, and validations for example. Or will it continue to be a fullstack framework focused on UI and "agnostic" to the specifics?

Regardless of the answer, thank you all for the hard work you have dedicated to this amazing framework! 🚀

**Comment by danielbergholz** [comment]
For context, a couple of articles about Remix possibly benefiting from becoming the new modern rails:

https://dev.to/remix-run-br/my-hopes-for-the-future-of-remix-13g4
https://martinliptak.medium.com/no-more-spas-4480dfb79b04

**Comment by seivan** [comment]
If it does, I sure hope Kysely becomes the query builder of choice. You don't need an ORM when you got static types.

**Comment by tombohub** [comment]
adonis is full stack framework, it doesn't have interactive frontend. If it can somehow integrate on top
